src/toil/provisioners/abstractProvisioner.py

```diff
@@ -469,26 +472,27 @@ def __init__(self):
             # Holds strings like "ssh-rsa actualKeyData" for keys to authorize (independently of cloud provider's system)
             self.sshPublicKeys = []
 
-        def addFile(self, path: str, owner: str = 'root', permissions: str = '0755', content: str = ''):
+        def addFile(self, path: str, filesystem: str = 'root', mode: int = int("0755", 8), contents: str = ''):
             """
-            Make a file on the instance with the given owner, permissions, and content.
+            Make a file on the instance with the given filesystem, mode (in decimal), and contents.
             """
 
-            self.files.append({'path': path, 'owner': owner, 'permissions': permissions, 'content': content})
+            contents = 'data:text/plain;charset=utf-8;base64,' + b64encode(contents.encode('utf-8')).decode('utf-8')
```

Comment:
It appear to me that base64 encoding the files actually takes less space. I think it's because the non-base64 encoded data URIs use three bytes to store special characters (e.g.: space is represented as `%20`), and we have a lot of spaces, line breaks, etc. Both are larger than the YAML equivalent though.

We should probably consider switching to the `https` or `s3` schemes and let Ignition fetch them during runtime, instead of writing them here.

Reply:
For posterity, we'll make Ignition fetch from cloud storage if the config is larger than the limit, because Flatcar only supports 2.x config versions, and this is their recommended setup.